src/HttpClientInterception/HttpInterceptionResponse.cs

```diff
@@ -7,6 +7,8 @@ namespace JustEat.HttpClientInterception;
 
 internal sealed class HttpInterceptionResponse
 {
+    internal Func<bool>? ConditionalMatcher { get; set; }
```

Comment:
While I don't think we need another one here and could just re-use `UserMatcher`, as-written the also wouldn't support the HTTP request being an input to the conditional matching, or any async evaluation.